newIDE/app/src/GameDashboard/MultiplayerAdmin.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+
+import * as React from 'react';
+import { Trans, t } from '@lingui/macro';
+import { ColumnStackLayout } from '../UI/Layout';
+import {
+  getLobbyConfiguration,
+  updateLobbyConfiguration,
+  type LobbyConfiguration,
+} from '../Utils/GDevelopServices/Play';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import Text from '../UI/Text';
+import { Line } from '../UI/Grid';
+import RaisedButton from '../UI/RaisedButton';
+import LeftLoader from '../UI/LeftLoader';
+import InfoBar from '../UI/Messages/InfoBar';
+import PlaceholderError from '../UI/PlaceholderError';
+import AlertMessage from '../UI/AlertMessage';
+import Link from '../UI/Link';
+import { getHelpLink } from '../Utils/HelpLink';
+import Window from '../Utils/Window';
+import SelectField from '../UI/SelectField';
+import SelectOption from '../UI/SelectOption';
+
+const defaultMaximumNumberOfPlayers = 4;
+const minimumNumberOfPlayers = 2;
+
+type Props = {|
+  gameId: string,
+|};
+
+const MultiplayerAdmin = ({ gameId }: Props) => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const [fetchingError, setFetchingError] = React.useState<React.Node>(null);
+  const [isSaving, setIsSaving] = React.useState<boolean>(false);
+  const [maxPlayersValue, setMaxPlayersValue] = React.useState<number>(2);
+  const {
+    getAuthorizationHeader,
+    profile,
+    limits,
+    subscription,
+  } = React.useContext(AuthenticatedUserContext);
+  const [
+    lobbyConfiguration,
+    setLobbyConfiguration,
+  ] = React.useState<?LobbyConfiguration>(null);
+  const [infoBarMessage, setInfoBarMessage] = React.useState<React.Node>(null);
+  const userId = profile ? profile.id : null;
+  const maximumNumberOfPlayers = limits
+    ? limits.capabilities.multiplayer.maxPlayersPerLobby
+    : defaultMaximumNumberOfPlayers;
+  const isUserOnFreePlan = !subscription || !subscription.planId;
+
+  React.useEffect(
+    () => {
+      if (lobbyConfiguration) {
+        setMaxPlayersValue(lobbyConfiguration.maxPlayers);
+        return;
+      }
+      if (limits) {
+        setMaxPlayersValue(limits.capabilities.multiplayer.maxPlayersPerLobby);
+        return;
+      }
+    },
+    [lobbyConfiguration, limits]
+  );
+
+  const maxPlayersSelectOptions = React.useMemo(
+    () => {
+      const options = new Array(maximumNumberOfPlayers - 1)
+        .fill(0)
+        .map((_, index) => (
+          <SelectOption
+            value={index + 2}
+            label={(index + 2).toString()}
+            shouldNotTranslate
+          />
+        ));
+      if (isUserOnFreePlan) {
+        options.push(
+          <SelectOption
+            value={maximumNumberOfPlayers + 1}
+            label={t`${(
+              maximumNumberOfPlayers + 1
+            ).toString()}+ (Available with a subscription)`}
+            disabled
+          />
+        );
+      }
```

Comment:
nit: I find the "isUserOnFreePlan" overly restrictive as a logic. We should rather always display option from 2 to the maximum, and disable those who are below the `maxPlayersPerLobby`.

Free plan or not is not important. What is important is what your capability "maxPlayersPerLobby" is.

Reply:
fixed